google/_async_resumable_media/requests/_request_helpers.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Shared utilities used by both downloads and uploads.
+
+This utilities are explicitly catered to ``requests``-like transports.
+"""
+
+
+import functools
+
+from google._async_resumable_media import _helpers
+from google.resumable_media import common
+
+import google.auth.transport._aiohttp_requests as aiohttp_requests
+import aiohttp
```

Comment:
requests is an optional extra, but due to the use of an aiohttp specific timeout, we have to make an explicit dependency.

Reply:
Given the conversation on https://github.com/googleapis/google-auth-library-python/issues/618 I think we should keep aiohttp an extra for the immediate term, since we aren't exposing users to it.

Reply:
Agree